File: frontend/src/types/multimodal.ts
export interface ImageAnalysisRequest {
  prompt: string;
  category?: string;
  target_cost?: string;
  image_description?: string;
  image_analysis?: Record<string, any>;
}

export interface ImageAnalysisResponse {
  success: boolean;
  image_analysis: Record<string, any>;
  enhanced_prompt: string;
  product_insights: string[];
  visual_elements: string[];
  color_scheme?: string;
  packaging_style?: string;
  target_audience_hints: string[];
  brand_name?: string;
  product_type?: string;
  market_positioning?: string;
  formulation_hints?: string[];
  error?: string;
  
  // Enhanced fields for comprehensive analysis
  product_category?: string;
  intended_use?: string;
  key_ingredients: string[];
  claims: string[];
  packaging_type?: string;
  packaging_size?: string;
  packaging_material?: string;
  target_audience?: string;
  brand_style?: string;
  competitor_positioning?: string;
  
  // New rich fields for comprehensive analysis
  formulation_insights: string[];
  consumer_insights: string[];
  price_positioning?: string;
  distribution_channels: string[];
  sustainability_aspects: string[];
  innovation_claims: string[];
  brand_story?: string;
  usage_instructions?: string;
  storage_requirements?: string;
  
  // Intent classification fields
  intent_classification?: {
    product_type_intent?: string;
    target_audience_intent?: string;
    benefit_intent?: string;
    ingredient_intent: string[];
    formulation_intent?: string;
    market_positioning_intent?: string;
    packaging_intent?: string;
    budget_intent?: string;
    competitor_intent?: string;
    sustainability_intent?: string;
    confidence_score?: number;
    primary_intent?: string;
  };
}

export interface MultiModalRequest {
  text_prompt: string;
  category?: string;
  target_cost?: string;
  image_analysis?: Record<string, any>;
  fusion_strategy: 'enhanced' | 'balanced' | 'image_primary';
}

export interface MultiModalResponse {
  success: boolean;
  fused_prompt: string;
  image_insights: Record<string, any>;
  text_enhancements: string[];
  formulation?: Record<string, any>;
  error?: string;
}

export interface ImageUploadResponse {
  success: boolean;
  image_url?: string;
  analysis?: Record<string, any>;
  error?: string;
}

export interface ImageAnalysisResult {
  product_type?: string;
  visual_elements: string[];
  color_scheme?: string;
  packaging_style?: string;
  target_audience_hints: string[];
  product_insights: string[];
  formulation_hints: string[];
  market_positioning?: string;
}

export interface FusionStrategy {
  id: 'enhanced' | 'balanced' | 'image_primary';
  name: string;
  description: string;
}

export const FUSION_STRATEGIES: FusionStrategy[] = [
  {
    id: 'enhanced',
    name: 'Enhanced Fusion',
    description: 'Combine text and image insights with equal weight for comprehensive analysis'
  },
  {
    id: 'balanced',
    name: 'Balanced Approach',
    description: 'Balance text requirements with visual elements from the image'
  },
  {
    id: 'image_primary',
    name: 'Image-First',
    description: 'Prioritize visual elements and styling from the image'
  }
]; 

File: backend/app/models/multimodal.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from fastapi import UploadFile

class ImageAnalysisRequest(BaseModel):
    """Request model for image analysis"""
    prompt: str
    category: Optional[str] = None
    target_cost: Optional[str] = None
    image_description: Optional[str] = None
    image_analysis: Optional[Dict[str, Any]] = None

class ImageAnalysisResponse(BaseModel):
    """Response model for image analysis"""
    success: bool
    image_analysis: Dict[str, Any]
    enhanced_prompt: str
    product_insights: List[str]
    visual_elements: List[str]
    color_scheme: Optional[str] = None
    packaging_style: Optional[str] = None
    target_audience_hints: List[str]
    brand_name: Optional[str] = None
    error: Optional[str] = None
    
    # Enhanced fields for comprehensive analysis
    product_category: Optional[str] = None
    intended_use: Optional[str] = None
    key_ingredients: List[str] = []
    claims: List[str] = []
    packaging_type: Optional[str] = None
    packaging_size: Optional[str] = None
    packaging_material: Optional[str] = None
    target_audience: Optional[str] = None
    brand_style: Optional[str] = None
    competitor_positioning: Optional[str] = None
    
    # New rich fields for comprehensive analysis
    formulation_insights: List[str] = []
    market_positioning: Optional[str] = None
    consumer_insights: List[str] = []
    price_positioning: Optional[str] = None
    distribution_channels: List[str] = []
    sustainability_aspects: List[str] = []
    innovation_claims: List[str] = []
    brand_story: Optional[str] = None
    usage_instructions: Optional[str] = None
    storage_requirements: Optional[str] = None
    
    # Intent classification fields
    intent_classification: Optional[Dict[str, Any]] = None

class MultiModalRequest(BaseModel):
    """Combined text and image analysis request"""
    text_prompt: str
    category: Optional[str] = None
    target_cost: Optional[str] = None
    image_analysis: Optional[Dict[str, Any]] = None
    fusion_strategy: str = "enhanced"  # "enhanced", "balanced", "image_primary"

class MultiModalResponse(BaseModel):
    """Response for multi-modal fusion"""
    success: bool
    fused_prompt: str
    image_insights: Dict[str, Any]
    text_enhancements: List[str]
    formulation: Optional[Dict[str, Any]] = None
    error: Optional[str] = None

class ImageUploadResponse(BaseModel):
    """Response for image upload and processing"""
    success: bool
    image_url: Optional[str] = None
    analysis: Optional[Dict[str, Any]] = None
    error: Optional[str] = None

class MultimodalSuggestion(BaseModel):
    prompt: str
    why: str
    how: str

class MultimodalSuggestionRequest(BaseModel):
    enhanced_prompt: str
    image_analysis: Dict[str, Any]
    category: Optional[str] = None

class MultimodalSuggestionResponse(BaseModel):
    suggestions: List[MultimodalSuggestion]
    success: bool = True
    message: str = "Multimodal suggestions generated successfully"
    error: Optional[str] = None 